cloud/linode/instances.go

```diff
@@ -284,12 +284,18 @@ func (i *instances) InstanceMetadata(ctx context.Context, node *v1.Node) (*cloud
 		return nil, err
 	}
 
-	uniqueAddrs := make(map[string]v1.NodeAddressType, len(node.Status.Addresses)+len(ips))
+	addresses := []v1.NodeAddress{{Type: v1.NodeHostName, Address: linode.Label}}
 	for _, ip := range ips {
-		if _, ok := uniqueAddrs[ip.ip]; ok {
+		addresses = append(addresses, v1.NodeAddress{Type: ip.ipType, Address: ip.ip})
+	}
```

Comment:
Adding ips first here as the ordering is already done in getInstanceAddresses() https://github.com/linode/linode-cloud-controller-manager/blob/main/cloud/linode/instances.go#L41-L63

Reply:
thanks for clarifying this, I was about to leave a comment asking about that 😄